File: PenetrationProjectAPI/src/main/java/com/ssafy/cafe/model/dto/Product.java
package com.ssafy.cafe.model.dto;

import java.util.Date;

import com.fasterxml.jackson.annotation.JsonFormat;

import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class Product {

	private Integer id;
	private String name;
	private String type;
	private Integer price;
	private String img;
	@JsonFormat(pattern="yyyy-MM-dd HH:mm:ss", timezone="Asia/Seoul")
	private Date date;
	private Integer views;
	
	@Builder
	public Product(Integer id, String name, String type, Integer price, String img, Date date, Integer views) {
		super();
		this.id = id;
		this.name = name;
		this.type = type;
		this.price = price;
		this.img = img;
		this.date = date;
		this.views = views;
	}

	@Builder
	public Product(Integer id, String name, String type, Integer price, String img) {
		super();
		this.id = id;
		this.name = name;
		this.type = type;
		this.price = price;
		this.img = img;
	}

	public Product(String name, String type, Integer price, String img) {
		this.name = name;
		this.type = type;
		this.price = price;
		this.img = img;
	}
}

File: SmartStoreApp/app/src/main/java/com/ssafy/smartstore/dto/Product.kt
package com.ssafy.smartstore.dto

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.io.Serializable

data class Product(var name: String?, var type: String, var price: Int, var img: String, var date: String, var views:Int) : Serializable {

    var id = 0

    constructor(id: Int, name: String, type: String, price: Int, img: String, date: String, views: Int)  : this(name, type, price, img, date, views) {
        this.id = id
    }

}